### ISSUE TITLE
Update submission process for post-review

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#5579]
Per https://docs.google.com/document/d/1rZvM2QGk8WtkNUDRvLEvrbqZ190hvHfOle4zVRllQmw/edit#:

The submission flow will have the following changes:

* Replace the Developer Agreement step with a step linking to the 3 main documents: Developer Agreement, Review Policy, and Review Rules. Each document will have a checkbox next to it indicating the developer has read them and agreed to them.
Ensure all developers see these documents on their next submission, even for an update.
* ~~For WebExtensions, a subset of flags raised by static validation will be shown after it completes. They are specified on this spreadsheet, in the column “Flag with developer during submission”.~~ Moved to #5753
* For WebExtensions, the text in the last step should be adjusted to reflect the new process.

This should be behind a waffle.

Currently blocked on UX/copy, though I may be able to just figure out something by building from the existing stuff.

**Comment by diox** [comment]
@jvillalobos can you clarify this:
> Ensure all developers see these documents on their next submission, even for an update.

Should developers see these documents on every update once this goes live, or just their next one (i.e. make sure they read this every time or just once because existing developers may not have read it yet) ?

> For WebExtensions, the text in the last step should be adjusted to reflect the new process.

Can I have more details on this ?

**Comment by jvillalobos** [comment]
> Should developers see these documents on every update once this goes live, or just their next one 

Just the next time they submit an add-on, either a new one or an update. We just want everyone to be aware of the existence of these documents.

> >  For WebExtensions, the text in the last step should be adjusted to reflect the new process.
>
> Can I have more details on this ?

The text in the first paragraph of the last submission step says `You're done!`, followed by a couple of sentences about waiting for review. For WebExtensions, those two sentences should be removed. The new message should read `You're done! This version is now available on our site.`.

> Also, this is what I have at the moment in terms of UX

Instead of embedding the doc like we did before, we just want to have a checkbox for each of the three documents. Something like this:

- [ ] [Developer Agreement](https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Agreement).
- [ ] [Review Policy](https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Reviews).
- [ ] Review Rules (link TBD).

**Comment by diox** [comment]
Regarding the message, at least at the beginning, "now available" won't be true (because the script to auto approve is executed once every hour). I suggest something vague like "will be available shortly" or "should be available within a couple hours".

**Comment by jvillalobos** [comment]
I mentioned in another issue that we want the approval to be immediate. I'd prefer that we do that before turning post-review on, and avoid having to go through multiple stages of review experience.

**Comment by diox** [comment]
As long as there are legacy add-ons needing the review process, I don't want to make approval immediate, because that adds complexity to the status workflow, edge cases, potential bugs, to an already fragile codebase. This is precisely one of the reasons why I chose to make auto approve a cronjob separate from this mess in the first place.

**Comment by muffinresearch** [comment]
@jvillalobos could we defer the requirement about aiming for immediate approvals for a separate discussion/issue? I think it would be best dealt with independently later on so that we don't complicate anything at this stage. It would be a good topic to discuss at SF if not before too.

**Comment by jvillalobos** [comment]
Since we'll continue to support legacy add-ons to some extent for about a year more, I really want to get this working correctly now rather than later. We can move this discussion elsewhere to figure out what will be needed.

**Comment by jvillalobos** [comment]
If the cron job isn't very expensive to run, can we consider running it every minute, or something like that? That way we all get what we want :)

**Comment by diox** [comment]
We may be able to move it to every 5 minutes or so, , but I'd like to see how it behaves with the regular queue and not just the updates queue first. Also, it's usually super fast, but the time slot was carefully chosen to avoid it running in parallel with other big tasks.

Note that in any case the approving will have to be done asynchronously, so it will *never* be instant. If there is load on the platform it could take a minute regardless of the implementation.

**Comment by jvillalobos** [comment]
A 5-10 minute wait for developers should be fine. The message can be tweaked to something like `You're done! This version will be available on our site shortly.`

**Comment by diox** [comment]
@jvillalobos any update on the Review Rules link ? I'll like to be done with this issue before SF if possible.

**Comment by jvillalobos** [comment]
Let's point to https://developer.mozilla.org/en-US/Add-ons/AMO/Policy/Rules (404 at present). @wagnerand, okay with you?

**Comment by wagnerand** [comment]
👍

**Comment by diox** [comment]
@jvillalobos : I had forgotten this:
> For WebExtensions, a subset of flags raised by static validation will be shown after it completes. They are specified on this spreadsheet, in the column “Flag with developer during submission”.

What should be shown exactly, with what message ? Note that the static analysis flags that are currently implemented (checks usage of `eval`, `document.write()`, `innerHTML`, etc) are already displayed as warnings at upload time.

**Comment by jvillalobos** [comment]
The idea is to show them in the submission flow page, so they are visible without the extra click to see the full results. The message would be the same as the one the validator displays, and should only appear once per type of flag. We used to have an "upload checklist" that was similar, but I think we removed it. @eviljeff might remember what I'm referring to.

**Comment by diox** [comment]
Ok, I can access the validation message easily, so this should be straightforward, thanks.

**Comment by diox** [comment]
Split that last part into a separate issue, #5753 to avoid polluting this one with more details.

**Comment by ValentinaPC** [comment]
Verified this on AMO-dev FF55(Win 7).
- the check-boxes were added to first step of submission while using a new account.
- the final step of webext submission was change accordingly (first submission and updates)
- everything is under a waffle
- implemented weights are working as expected, those who still need implementation will be tracked separately.

I have a question:
@diox : shouldn't we use same names in description, links or button labels? To exclude confusions.
What do you think?
![2017-09-08_1543](https://user-images.githubusercontent.com/15685960/30212332-27250790-94ad-11e7-9363-48f8ade9bfba.png)

**Comment by diox** [comment]
Good call. @jvillalobos can you confirm that the current copy is ok or give an updated one for the text + checkboxes label + button?

**Comment by jvillalobos** [comment]
Yes, good catch. The checkboxes are the ones that need to be updated. They should say `Firefox Add-on Distribution Agreement`, and `Review Policies and Rules`.

**Comment by diox** [comment]
Ah I had changed it in https://github.com/mozilla/addons-server/issues/6293 because of the issue description :) I'll revert that :)

**Comment by ValentinaPC** [comment]
Verified as fixed on AMo-dev FF55(Win 7).
![2017-09-12_1025](https://user-images.githubusercontent.com/15685960/30313142-fdb04f32-97a4-11e7-856f-baa239607302.png)
